### ISSUE TITLE
Cube backends: two-sided clamp maps NaN to a bound while clamp_min/clamp_max propagate it

### ISSUE THREAD

**Issue by Liberxue** [tracel-ai/burn#5659]
**Describe the bug**

On the cube backends, two-sided `Tensor::clamp` replaces NaN with a bound instead of propagating it. The one-sided `clamp_min` and `clamp_max` on the same backend are correct, so the three disagree with each other.

**To Reproduce**

Verified on CUDA (`--features cuda`), for both F32 and F64:

```rust
let tensor = TestTensor::<1>::from([f32::NAN, -1.0, 2.0]).cast(dtype);

tensor.clone().clamp_min(0.0)   // [NaN, 0.0, 2.0]   correct
tensor.clone().clamp_max(1.0)   // [NaN, -1.0, 1.0]  correct
tensor.clamp(0.0, 1.0)          // [1.0,  0.0, 1.0]  NaN became the upper bound
```

`torch.clamp(nan, 0, 1)` returns NaN.

**Analysis**

`burn-cubecl` does not override `float_clamp_min` / `float_clamp_max`, so they fall back to the trait default, which masks with `float_lower_elem` / `float_greater_elem` — false for NaN, so NaN survives.

It does override the two-sided op (`crates/burn-cubecl/src/ops/tensor.rs:798`), which dispatches to `kernel::clamp` (`crates/burn-cubecl/src/kernel/clamp.rs:26`) and from there to `cubecl::prelude::clamp`. That lowering is where the NaN is lost, so the fix likely belongs in cubecl rather than in burn.

**Expected behavior**

`clamp`, `clamp_min` and `clamp_max` agree with each other and with PyTorch: NaN in, NaN out.

**Desktop**

- OS: Ubuntu 26.04 LTS, x86_64
- burn: `09f9b3f78b9210fd91f64c96dae50950bc16b7d4`
- CUDA 12.4, Tesla V100
- rustc 1.97.1

**Additional context**

Same class of bug as #5609, which covered burn-flex. While preparing #5658 I ran the NaN assertions across three backends:

| op | flex (after #5658) | ndarray | cuda |
| --- | --- | --- | --- |
| `relu` | pass | pass | pass |
| `clamp_min` | pass | pass | pass |
| `clamp_max` | pass | pass | pass |
| `clamp` | pass | pass | **fail** |

Because of this, #5658 leaves `clamp_nan_propagation` in `burn-backend-tests` gated to `flex` / `ndarray`. Once this is fixed, that gate can be removed so the assertion covers every backend, alongside the existing ungated `clamp_min` / `clamp_max` ones. #5615 tracks a separate question about the clamp bounds themselves.

I have not checked wgpu, metal, vulkan or rocm, but they share the same cubecl lowering, so they are likely affected too.
